tests/test_dask_imread.py

```diff
@@ -1,13 +1,86 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
+
 from __future__ import absolute_import
 
+import numbers
+
 import pytest
 
+import numpy as np
+from skimage.external import tifffile
+
+import dask.array.utils as dau
+import dask_imread
+
+
+@pytest.mark.parametrize(
+    "err_type, nframes",
+    [
+        (ValueError, 1.0),
+        (ValueError, 0),
+        (ValueError, -1),
+    ]
+)
+def test_errs_imread(err_type, nframes):
+    with pytest.raises(err_type):
+        dask_imread.imread("test.tiff", nframes=nframes)
+
+
+@pytest.mark.parametrize(
+    "seed",
+    [
+        0,
+        1,
+    ]
+)
+@pytest.mark.parametrize(
+    "nframes, shape",
+    [
+        (1, (1, 4, 3)),
+        (3, (1, 4, 3)),
+        (1, (5, 4, 3)),
+        (2, (5, 4, 3)),
+        (1, (10, 5, 4, 3)),
+        (5, (10, 5, 4, 3)),
+        (10, (10, 5, 4, 3)),
+    ]
+)
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        np.int16,
+        np.int32,
+        np.float32,
+    ]
+)
+def test_tiff_imread(tmpdir, seed, nframes, shape, dtype):
```

Comment:
Might be worth testing other kinds of image data. However am unclear on what other types might be of interest. Have opened issue ( https://github.com/dask-image/dask-imread/issues/8 ) to collect use cases.